crates/core/tedge/src/system_services/managers/general_manager.rs

```diff
@@ -0,0 +1,268 @@
+use crate::system_services::{
+    CommandBuilder, SystemConfig, SystemConfigError, SystemService, SystemServiceError,
+    SystemServiceManager,
+};
+use itertools::Itertools;
+use std::fmt;
+use std::path::PathBuf;
+use std::process::ExitStatus;
+use tedge_users::{UserManager, ROOT_USER};
+
+#[derive(Debug)]
+pub struct GeneralServiceManager {
+    user_manager: UserManager,
+    system_config: SystemConfig,
+}
+
+impl GeneralServiceManager {
+    pub fn try_new(
+        user_manager: UserManager,
+        config_root: PathBuf,
+    ) -> Result<Self, SystemConfigError> {
+        let system_config = SystemConfig::new(config_root);
+        Ok(Self {
+            user_manager,
+            system_config,
+        })
+    }
+}
+
+impl SystemServiceManager for GeneralServiceManager {
+    fn name(&self) -> &str {
+        &self.system_config.name
+    }
+
+    fn check_operational(&self) -> Result<(), SystemServiceError> {
+        let exec_command = ExecCommand::try_new(self.system_config.is_available.clone())?;
+
+        match exec_command.to_command().status() {
+            Ok(status) if status.success() => Ok(()),
+            _ => Err(SystemServiceError::ServiceManagerUnavailable(
+                self.name().to_string(),
+            )),
+        }
+    }
+
+    fn stop_service(&self, service: SystemService) -> Result<(), SystemServiceError> {
+        let config = replace_with_service_name(&self.system_config.stop, service)?;
+        let exec_command = ExecCommand::try_new(config)?;
+        self.run_service_command_as_root(exec_command)?
+            .must_succeed()
+    }
+
+    fn restart_service(&self, service: SystemService) -> Result<(), SystemServiceError> {
+        let config = replace_with_service_name(&self.system_config.restart, service)?;
+        let exec_command = ExecCommand::try_new(config)?;
+        self.run_service_command_as_root(exec_command)?
+            .must_succeed()
+    }
```

Comment:
Try to avoid duplicated code between all these commands with a `run_service_command(service: SystemService, command: Vec<String>)` function.